### DISCUSSION TITLE
UTF-8 chars in VScode debug console

### DISCUSSION THREAD

**Discussion by pmudry** [scalameta/metals#2591] [Q&A] [answered]
I can't get the debug console display UTF-8 chars in VS code. To replicate : 

```scala
object Trivial extends App {
  println("éléphant")
}
```

Ouput is:
```
�l�phant
```

Does everyone gets the same result ?

_Edited_ : the output is wrong using both the `run lens` in VSCode and with a launch configuration in `launch.json`. However, there is a solution thanks to @tgodzik below !

**Comment by tgodzik** [comment, marked as answer]
Thanks for the question! It seems to work fine for me. Maybe it's related to encoding in VS Code? Could you search the settings for `encoding` and see if it's properly set there? Alternatively, you could try adding `-Dfile.encoding=UTF-8` to the jvm options in the run configuration (`.vscode/launch.json`), but I don't need to set it myself and it seems to work.

**Reply by pmudry** [reply]
Hi ! As suggested, I added the encoding option to JVM argument and it now works 👍 
For reference, my `launch.json` 

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "scala",
            "request": "launch",
            "name": "Simple",
            "mainClass": "series.serie5.for_postcard",
            "args": [],
            "jvmOptions": ["-Dfile.encoding=UTF-8"],
            "env": {}
        }
    ]
}
```

Thank you so much for your help!

**Reply by tgodzik** [reply]
Though I wonder, why is that really needed :thinking: I've never had issues with encoding, maybe it's also possible to set it system wide for java? If anyone knows, do let us know :sweat_smile:

**Reply by pmudry** [reply]
Me neither, that's odd. 

In addition, running via the `run lens` in VScode exhibits the same behavior in debug console. This is not expected either and adding `.jvmopts` does not change anything.

**Reply by JonaSi754** [reply]
I have the same problem in C++, I wonder how to solve it. I need it because I want to show some Chinese characters in console

**Reply by tgodzik** [reply]
Not sure, maybe something like https://stackoverflow.com/questions/11646368/how-to-set-file-encoding-format-to-utf8-in-c ?

**Reply by leprinco** [reply]
thanks so much, this helped me a lot. Just a small note, seems in my 2022 version of VS Code jvmOptions is replaced by vmArgs :
              ...
            "args": "",
            **"vmArgs": ["-Dfile.encoding=UTF-8"],**
            "classPaths": []
               ...
